From 6c7160be3169514aeecf23e423dfe2fe29c1b5e9 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Sun, 10 Jan 2016 00:02:00 +0800
Subject: [PATCH 0834/5242] MLK-12203-1 ARM: imx: adjust i.MX7D DDR retention
 mode for LPSR

commit  a5dd40191b48cff48ffc5d88cabf0976f7c5ecaf from
https://source.codeaurora.org/external/imx/linux-imx.git

Per design team's recommendation, for i.MX7D TO1.1
LPSR mode, as IOMUXC will lost power, so it needs to
use TO1.0's flow to avoid CKE toggle during retention,
but it has a limitation of POR reset fail during LPSR.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/suspend-imx7.S |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/suspend-imx7.S b/arch/arm/mach-imx/suspend-imx7.S
index df78a929..074efdc 100644
--- a/arch/arm/mach-imx/suspend-imx7.S
+++ b/arch/arm/mach-imx/suspend-imx7.S
@@ -309,7 +309,12 @@
 	ldr	r11, [r0, #PM_INFO_MX7_IOMUXC_GPR_V_OFFSET]
 	ldr	r7, =0x38000000
 	str	r7, [r11]
-	b	11f
+
+	/* LPSR mode need to use TO1.0 flow as IOMUX lost power */
+	ldr	r10, [r0, #PM_INFO_MX7_LPSR_V_OFFSET]
+	ldr	r7, [r10]
+	cmp	r7, #0x0
+	beq	11f
 10:
 	/* reset ddr_phy  */
 	ldr	r11, [r0, #PM_INFO_MX7_ANATOP_V_OFFSET]
-- 
1.7.9.5

